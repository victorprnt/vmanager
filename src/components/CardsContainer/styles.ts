import styled from 'styled-components'

export const Wrapper = styled.section`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  gap: 1vw;

  width: 100%;
  max-height: 100%;
  margin: 1vw 1vw 1vw 4vw;

  .infocard-container {
    display: flex;
    gap: 2vw;
    justify-content: center;

    width: 100%;
  }
`
