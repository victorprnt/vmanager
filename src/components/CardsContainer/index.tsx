import GraphCard from '../GraphCard'
import InfoCard from '../InfoCard'
import * as S from './styles'

const CardsContainer = () => (
  <S.Wrapper>
    <div className="infocard-container">
      <InfoCard aria-label="infocard" />
      <InfoCard aria-label="infocard" />
      <InfoCard aria-label="infocard" />
    </div>
    <GraphCard aria-label="graphcard" />
    <GraphCard aria-label="graphcard" />
  </S.Wrapper>
)

export default CardsContainer
