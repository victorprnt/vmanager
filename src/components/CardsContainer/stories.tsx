// import { Story, Meta } from '@storybook/react'
import CardsContainer from '.'

// export default {
//   title: 'CardsContainer',
//   component: CardsContainer
// } as Meta

// export const Default: Story = () => <CardsContainer />
