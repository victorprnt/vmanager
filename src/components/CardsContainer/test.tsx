import { render, screen } from '@testing-library/react'

import CardsContainer from '.'

describe('<CardsContainer />', () => {
  it('should render the main portion of the cards', () => {
    const { container } = render(<CardsContainer />)

    // expect(screen.getAllByLabelText(/infocard/i)).toHaveLength(3)
    // expect(screen.getAllByLabelText(/graphcard/i)).toHaveLength(2)

    expect(container.firstChild).toMatchSnapshot()
  })
})
