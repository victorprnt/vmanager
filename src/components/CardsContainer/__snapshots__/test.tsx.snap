// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardsContainer /> should render the main portion of the cards 1`] = `
<section
  class="sc-dkzDqf grYqif"
>
  <div
    class="infocard-container"
  >
    <div
      class="sc-gsnTZi ggXPc infocard"
    >
      <img
        alt="Infocard Image"
        class="infocard"
        src=""
      />
      <div
        class="infocard-text"
      >
        <p
          class="infocard-value"
        >
          Value
        </p>
        <p
          class="infocard-label"
        >
          Label
        </p>
      </div>
    </div>
    <div
      class="sc-gsnTZi ggXPc infocard"
    >
      <img
        alt="Infocard Image"
        class="infocard"
        src=""
      />
      <div
        class="infocard-text"
      >
        <p
          class="infocard-value"
        >
          Value
        </p>
        <p
          class="infocard-label"
        >
          Label
        </p>
      </div>
    </div>
    <div
      class="sc-gsnTZi ggXPc infocard"
    >
      <img
        alt="Infocard Image"
        class="infocard"
        src=""
      />
      <div
        class="infocard-text"
      >
        <p
          class="infocard-value"
        >
          Value
        </p>
        <p
          class="infocard-label"
        >
          Label
        </p>
      </div>
    </div>
  </div>
  <div
    class="sc-bczRLJ iZVTKG"
  >
    <h2
      class="graph-title graph-text"
    >
      Graph Title
    </h2>
    <h3
      class="graph-subtitle graph-text"
    >
      subtitle
    </h3>
    <img
      alt="Graph image"
      src=""
    />
  </div>
  <div
    class="sc-bczRLJ iZVTKG"
  >
    <h2
      class="graph-title graph-text"
    >
      Graph Title
    </h2>
    <h3
      class="graph-subtitle graph-text"
    >
      subtitle
    </h3>
    <img
      alt="Graph image"
      src=""
    />
  </div>
</section>
`;
